src/api/v2/compatibility/CompatibleBranchDataProvider.ts

```diff
@@ -4,21 +4,29 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzExtParentTreeItem, AzExtTreeDataProvider, AzExtTreeItem, IFindTreeItemContext, ISubscriptionContext, ITreeItemPickerContext } from '@microsoft/vscode-azext-utils';
-import { AppResource, AppResourceResolver } from '@microsoft/vscode-azext-utils/hostapi';
+import type { AppResource, AppResourceResolver } from '@microsoft/vscode-azext-utils/hostapi';
 import * as vscode from 'vscode';
+import type { ApplicationResource, BranchDataProvider, ResourceModelBase } from '../../../api/v2/v2AzureResourcesApi';
 import { createSubscriptionContext } from '../../../utils/v2/credentialsUtils';
-import { ApplicationResource, BranchDataProvider, ResourceModelBase } from '../v2AzureResourcesApi';
+import { CompatibleResolvedApplicationResourceTreeItem } from './CompatibleApplicationResourceTreeItem';
 
+/**
+ * Provides compatibility between an `AppResourceResolver` (v1) and a `BranchDataProvider` (v2)
+ */
 export class CompatibleBranchDataProvider<TResource extends ApplicationResource, TModel extends AzExtTreeItem & ResourceModelBase> extends AzExtTreeDataProvider implements BranchDataProvider<TResource, TModel> {
     private readonly overrideOnDidChangeTreeDataEmitter = new vscode.EventEmitter<TModel | undefined>();
 
     public constructor(private readonly resolver: AppResourceResolver, loadMoreCommandId: string) {
-        super(undefined as unknown as AzExtParentTreeItem, loadMoreCommandId);
+        // Using `{}` here so property assignment doesn't throw
+        super({} as unknown as AzExtParentTreeItem, loadMoreCommandId);
     }
 
     //#region TreeDataProvider
 
-    public override readonly onDidChangeTreeData = this.overrideOnDidChangeTreeDataEmitter.event;
+    // @ts-expect-error `getParent` is not meant to be defined by `BranchDataProvider`s but is already defined by `AzExtTreeDataProvider`
+    public override get onDidChangeTreeData(): Event<AzExtTreeItem | undefined> {

```

Comment:
What made this change necessary?

Reply:
It's being set in the constructor of `AzExtTreeDataProvider`. I changed it back to the getter, and added an empty setter.

Reply:
[I'm not seeing it](https://github.com/microsoft/vscode-azuretools/blob/main/utils/src/tree/AzExtTreeDataProvider.ts#L30), is there some indirect thing? It doesn't look like it ever does a set of `onDidChangeTreeData`.

Reply:
Sorry, I thought I saw it being set. Well it's being set somewhere, let me find out where. Also TS is unhappy that it's a property in `AzExtTreeDataProvider`, but is overridden here as an accessor.

Reply:
But if I remove `readonly` it'll work, but I'm not sure we should allow it to be set from elsewhere. Which is why I went with the empty setter.

Reply:
Ugh. TypeScript is so fussy about this. I think an empty setter is fine here.